### PR TITLE
Add advanced setting to allow more seek accuracy

### LINKED ISSUE #744: Seeking is inaccurate in some MP3 files
### Description

Seeking is inaccurate for variable bitrate (VBR) MP3 files. This is a [known ExoPlayer limitation](https://exoplayer.dev/troubleshooting.html#why-is-seeking-inaccurate-in-some-mp3-files). 

For exact seeking in such files, we can set `Mp3Extractor#FLAG_ENABLE_INDEX_SEEKING` flag on [DefaultExtractorsFactory](https://exoplayer.dev/customization.html#customizing-extractor-flags), but it doesn't scale well for large MP3 files. With the flag enabled, it took us [3-8 secs to seek to end of the file](https://github.com/Automattic/pocket-casts-android/pull/733#issuecomment-1406886043) for the first time. 

We're keeping the existing behavior with optimization for speed over accuracy, as it fulfills a large majority of the users. This issue is created to track incidents where users report inaccurate seeking behavior and we identify in our troubleshooting that the file they're trying is a large MP3 file, and dynamic ads are not enabled for them (see related issue: https://github.com/Automattic/pocket-casts-android/issues/157).

### Step-by-step reproduction instructions

1. Play a large episode in MP3 format, e.g. https://pca.st/xrfsu987 
2. Seek forward/ backward a few times.
3. Seek to almost the end of the episode until a few secs are left for it to complete.
4. Listen till the end of the episode.
5. Notice that the episode keeps playing for a few more seconds than the original duration.

Also, note that the issue is not limited to episode playback overshooting actual duration, but also inaccurate seeking in-between the file.

### Screenshots or screen recording

Original duration | Episode plays beyond duration 
--|--
duration | issue

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### PR DESCRIPTION
**Pull request by ashiagr**
## Description

Now that we have an `Advanced Settings` section, I'm re-proposing an option to allow more seek accuracy (earlier PR: https://github.com/Automattic/pocket-casts-android/pull/733).

It is not enabled by default, as [more accuracy comes at the cost of slower seeking](https://developer.android.com/media/media3/exoplayer/troubleshooting#why-is-seeking-inaccurate-in-some-mp3-files), particularly for large mp3 files. However, with this advanced setting, and with a stronger warning as advised here: https://github.com/Automattic/pocket-casts-android/pull/733#discussion_r1089262870), users will have a choice to prioritize accuracy over speed.

Fixes #744

## Review Instructions

Requires at least two approvals to merge this PR.

## Testing Instructions

Steps to reproduce

0. Disable `Beta features` -> `Cache playing episode`, kill and restart the app
1. Play a long episode in MP3 format, e.g. https://pca.st/xrfsu987
3. Seek to almost the end of the episode until a few secs are left for it to complete
4. Listen till the end of the episode
5. ✅ Notice that the episode keeps playing for a few more seconds than the original duration.

Test new behavior

7. Go to `Profile` -> `Settings` -> `Advanced Settings`
8. Enable "Prioritize seek accuracy"
9. Reopen the same episode. 
10. Repeat steps 1-3
11. ✅ Notice that the episode ends at the original duration.

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md [TO BE ADDED]
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ashiagr** [comment]
@mebarbosa, @MiSikora 

I want one more approval for this PR. Can anyone of you do the favor?

**Review by MiSikora — APPROVED** [review]

**Review by mebarbosa — APPROVED** [review]
Awesome work! ❤️ 

I also tried to combine different tests such as skipping last and all worked as expected

**Pull request merged**
